packages/labs/virtualizer/src/Virtualizer.ts

```diff
@@ -764,42 +776,44 @@ export class Virtualizer {
     );
   }
 
-  private _layoutCompleteResolver: Function | null = null;
-  private _layoutCompleteRejecter: Function | null = null;
-  private _pendingLayoutComplete: number | null = null;
   public get layoutComplete(): Promise<void> {
-    return new Promise((resolve, reject) => {
-      this._layoutCompleteResolver = resolve;
-      this._layoutCompleteRejecter = reject;
-    });
+    // Lazily create promise
+    if (!this._layoutCompletePromise) {
+      this._layoutCompletePromise = new Promise((resolve, reject) => {
+        this._layoutCompleteResolver = resolve;
+        this._layoutCompleteRejecter = reject;
+      });
+    }
+    return this._layoutCompletePromise!;
   }
+
   private _rejectLayoutCompletePromise(reason: string) {
     if (this._layoutCompleteRejecter !== null) {
       this._layoutCompleteRejecter!(reason);
     }
     this._resetLayoutCompleteState();
   }
+
   private _scheduleLayoutComplete() {
-    if (this._pendingLayoutComplete !== null) {
-      cancelAnimationFrame(this._pendingLayoutComplete);
+    // Don't do anything unless we have a pending promise
+    // And only request a frame if we haven't already done so
+    if (this._layoutCompletePromise && this._pendingLayoutComplete === null) {
+      // Wait one additional frame to be sure the layout is stable
+      this._pendingLayoutComplete = requestAnimationFrame(() =>
+        requestAnimationFrame(() => this._resolveLayoutCompletePromise())
```

Comment:
Interesting! I like the locality/cleanliness of expressing the extra nesting RAF here.  It appears to mean that resolving the layout complete promise is no longer deferrable past the second RAF, however, since `this._resolveLayoutCompletePromise()`.  I am trying to contrive a scenario where that would pose a risk...

Considering a case where _childrenSizeChanged() is called due to observation and that triggers a subsequent scheduleLayoutComplete but the subsequent one doesn't wait yet-another two animation frames--  I'll tell you what I'll make an issue to add more cases to the layout complete tests to verify expectations around cascading resize effects and the layoutComplete promise as a separate issue and approve this as is.

Reply:
Yeah, a full set of `layoutComplete` tests would be great. I *think* any given cycle of child measurements and re-layouts is going to occur within a single frame, so believe this logic is ok—but we should certainly confirm via tests.